src/shapes/tests/test_rectangle.py

```diff
@@ -247,7 +247,7 @@ def test08_differentiable_surface_interaction_ray_forward_follow_shape(variants_
     dr.forward(theta)
 
     assert dr.allclose(dr.grad(si.t), 0.0)
-    assert dr.allclose(dr.grad(si.p), 0.0)
+    assert dr.allclose(dr.grad(si.p), 0.0, atol=1e-6)
```

Comment:
Why the tighter tolerance here specifically?

Reply:
That's a leftover from a debugging session involving a polarized variant, removed -- thanks for catching that!

Reply:
Actually, this was necessary after all. The variant `llvm_ad_rgb_polarized_double` fails this test -- there is a small amount of error in that variant, and the allclose() check turns out to be too strict and it fails.